lib/services/local/dynamic_access.go

```diff
@@ -158,6 +176,138 @@ func (s *DynamicAccessService) UpsertAccessRequest(ctx context.Context, req serv
 	return nil
 }
 
+// GetPluginData loads all plugin data matching the supplied filter.
+func (s *DynamicAccessService) GetPluginData(ctx context.Context, filter services.PluginDataFilter) ([]services.PluginData, error) {
+	switch filter.Kind {
+	case services.KindAccessRequest:
+		data, err := s.getAccessRequestPluginData(ctx, filter)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return data, nil
+	default:
+		return nil, trace.BadParameter("unsupported resource kind %q", filter.Kind)
+	}
+}
+
+func (s *DynamicAccessService) getAccessRequestPluginData(ctx context.Context, filter services.PluginDataFilter) ([]services.PluginData, error) {
+	// Filters which specify Resource are a special case since they will match exactly zero or one
+	// possible PluginData instances.
+	if filter.Resource != "" {
+		item, err := s.Get(ctx, pluginDataKey(services.KindAccessRequest, filter.Resource))
```

Comment:
Not sure I understand `filter` if filter is just a resource name, why not make it consistent with the rest of the API?

Reply:
The `filter` also encodes resource kind and plugin name, it just isn't obvious reading this function body.  Resource kind is checked by the top-level `GetPluginData` function, and plugin name is used internally by the call to `filter.Match(...)`.